### ISSUE TITLE
Allow custom merge in AggregatingMergeTree when column is of simple type

### ISSUE THREAD

**Issue by asafcombo** [ClickHouse/ClickHouse#26253]
**Use case**
The current behavior of `AggregatingMergeTree` for simple columns is to do an `any` merge ( although according to my tests it looks like `anyFirst` ).

I suggest to allow a custom setting in the table settings to allow other function as well - for example `max`.

This will allow a more simple schema design that is more easy to maintain and read. See a more complex example later.

So let's look at the below

```sql
create database tests;
-- 
CREATE TABLE tests.example
(
    `timestamp` DateTime Codec (DoubleDelta, LZ4),
    `ne`                 Int32,
    `request_id`        String,

    `field_a`       String

)
ENGINE = AggregatingMergeTree()
    PARTITION BY (ne)
    ORDER BY (ne, timestamp, request_id);

-- 

insert into tests.example values ('2021-06-21T01:00:00', 5, '0-0-0-1', '');  -- first insert of key
insert into tests.example values ('2021-06-21T01:00:00', 5, '0-0-0-1', 'a_2');  -- update of key
-- 
optimize table  tests.example final;
-- 
select * from tests.example;
-- '2021-06-21T01:00:00' |  5 | '0-0-0-1' | ''
```

`timestamp, ne, request_id` are primary key fields and exists in each and every insert, however `field_a` is an updateable field and only exists on some of the inserts.  As its type is a simple String - the merge could either get the value 'a_2' or could get us the default value of ''.

The normal thing to do is to declare `field_a` as such:

```sql
    `field_a`       SimpleAggregateFunction(max, String),
```

however it causes the schema to be more complex than it has to be, and also causes some downstream issues ( like querying ion superset ).

With the current behavior CH  is handling it behind the scenes like: 
```sql
    `field_a`       SimpleAggregateFunction(any, String),
```

This will really come in handy for my use case where we have a few streams that needs to be merged. All streams know what the primary key is, but the first stream ( main stream )  has around 100 fields that exist only there.  It is not possible to append these fields to other streams as it is both redundant and is costly.

However for the secondary streams  only 1,2 fields are updated or appended, and we could use either explicit 
`SimpleAggregateFunction` or `AggregateFunction`  according to what we want to achieve. 

**Describe the solution you'd like**
I propose to add a table setting `default_agg_merge` that would work only on columns that are of a simple type that will allow to customize to default action when merging these columns.

**Comment by UnamedRus** [comment]
>  I propose to add a table setting default_agg_merge that would work only on columns that are of a simple type that will allow to customize to default action when merging these columns.

Currently table level settings isn't being replicated at most (except index_granularity*) AFAIK.

So it's possible to have different values on each replica which would lead to merges inconsistency.

**Comment by akuzm** [comment]
This would be just another way to do the same thing as `SimpleAggregateFunction`, but specifying it in a different way. Not sure we would want to support it.

Maybe there are some ways to improve the usability of SimpleAggregateFunction instead? What is the primary problem, downstream systems not understanding the unusual column type, and having to apply `finalizeAggregation`? For the latter, we could probably add some more syntactic sugar, e.g. make it apply `finalizeAggregation` transparently when you write the column name.

**Comment by asafcombo** [comment]
@akuzm yes it would definitely help downstream systems, but the main goal is to simplify the schema maintenance. 

As an example, a popular design is to use a `NULL TABLE -> AggregateMergeTree` table. 

Instead of having to keep track of all fields in both tables ( but with different types, the null table will have a String for example, where the downstream table will have a  `SimpleAggregateFunction`), we could keep just one schema, or atleast bookeep the fields and keep the same type.
